frontend/shared/src/utils/date.utils.ts

```diff
@@ -31,7 +32,21 @@ export const formatDate = (
   }).trim();
 };
 
-export const parseDate = (date: string, format = 'dd.MM.yyyy'): Date =>
-  parse(date, format, new Date());
+// TODO: remove format argument
```

Comment:
> Some comments, and I didnt read this through very precisely, just added some comments here and there... Generally I feel like it's getting harder to jump in to the application logic as there is lot of repetition and long files. Perhaps some refactoring would help, but I understand that if tight on schedule and no tests to protect against regeression bugs then refactoring something to avoid. But for outsider the code gradually gets harder to read and maintain. Also the business logic seem to be quite complex :)

I agree! It's getting more complex to comprehend how the application works, and what I am doing here as a first step is extracting some logic to separate reusable modules so they can be refactored, optimized, and tested.

Reply:
Cool, I just left it here in this PR to avoid modifying a lot of files.